.github/workflows/promote_release.yaml

```diff
@@ -0,0 +1,191 @@
+          echo "PR_MAIN_REF=$(git rev-list -n 1 ${PR_BRANCH}^^)" >> $GITHUB_ENV
+
+      - name: 'Check for PR/main divergence'
+        if: ${{ inputs.command == 'promote-release' }}
+        env:
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+        run: |
+          if [[ "${MAIN_HEAD}" != "${PR_MAIN_REF}" ]];
+          then
+            gh pr comment --body "Release PR (expects main ${PR_MAIN_REF}) has diverged from main (${MAIN_HEAD}). You must drop this release and restage the release." ${{ inputs.release-pr-issue-number }}
+            exit 1
+          fi
+
+      - name: 'Check build workflow has completed successfully'
+        if: ${{ inputs.command == 'promote-release' }}
+        env:
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+        run: |
+          STATUS=$(gh api --method POST \
+              -H "Accept: application/vnd.github+json" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              /repos/${{ github.repository }}/commits/${PR_MAIN_REF}/check-runs \
+              | jq --exit-status -r '.check_runs[] | select(.name=="build") | .status + "/" + .conclusion')
+
+          if [[ ${STATUS} != "completed/success" ]]; then
+            gh pr comment --body "Build workflow has not completed successfully (${STATUS}). Ask me again later." ${{ inputs.release-pr-issue-number }}
+            exit 1
+          fi
+
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: 'Set up Java'
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: temurin
+          server-id: ossrh
+          server-username: SONATYPE_TOKEN_USERNAME # env variable for Sonatype username
+          server-password: SONATYPE_TOKEN_PASSWORD # env variable for Sonatype password
+          overwrite-settings: true
+
+      - name: 'Transition staging repository to desired state'
+        env:
+          SONATYPE_TOKEN_USERNAME: ${{ vars.KROXYLICIOUS_SONATYPE_TOKEN_USERNAME }}
+          SONATYPE_TOKEN_PASSWORD: ${{ secrets.KROXYLICIOUS_SONATYPE_TOKEN_PASSWORD }}
+        run: ./scripts/transition-staging-repository-state.sh -s ${{ inputs.command == 'promote-release' && 'release' || 'drop' }}
+
+      - name: 'Unblock the merge'
+        if: ${{ inputs.command == 'promote-release' }}
+        env:
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+        run: |
+          gh api --method POST \
+                 -H "Accept: application/vnd.github+json" \
+                 -H "X-GitHub-Api-Version: 2022-11-28" \
+                 /repos/${{ github.repository }}/statuses/${PR_MAIN_REF} \
+                 -f "state=success" \
+                 -f "description=Sonatype Release Is Done" \
```

Comment:
```suggestion
                 -f "description=Release to Maven Central is complete" \
```

Reply:
Yes I think this change is worth applying